protocol/contracts/templegold/TempleGold.sol

```diff
@@ -324,20 +362,39 @@ import { TempleMath } from "contracts/common/TempleMath.sol";
         address /*_executor*/, // @dev unused in the default implementation.
         bytes calldata /*_extraData*/ // @dev unused in the default implementation.
     ) internal virtual override {
-        // @dev The src sending chain doesnt know the address length on this chain (potentially non-evm)
-        // Thus everything is bytes32() encoded in flight.
-        address toAddress = _message.sendTo().bytes32ToAddress();
-        // @dev Credit the amountLD to the recipient and return the ACTUAL amount the recipient received in local decimals
-        uint256 amountReceivedLD = _credit(toAddress, _toLD(_message.amountSD()), _origin.srcEid);
-
         /// @dev Disallow further execution on destination by ignoring composed message
         if (_message.isComposed()) { revert CannotCompose(); }
 
-        emit OFTReceived(_guid, _origin.srcEid, toAddress, amountReceivedLD);
+        if (_message.sendTo().bytes32ToAddress() == address(0)) {
+            /// @dev no need to burn, that happened in source chain
+            // already checked destination Eid for burn case in `send`
+            // update circulating supply
+            // _origin.sender is spice auction
+            _updateCirculatingSupply(_origin.sender.bytes32ToAddress(), _message.amountSD());
+        } else {
+            // @dev The src sending chain doesnt know the address length on this chain (potentially non-evm)
+            // Thus everything is bytes32() encoded in flight.
+            address toAddress = _message.sendTo().bytes32ToAddress();
+            // @dev Credit the amountLD to the recipient and return the ACTUAL amount the recipient received in local decimals
+            uint256 amountReceivedLD = _credit(toAddress, _toLD(_message.amountSD()), _origin.srcEid);
+
+            emit OFTReceived(_guid, _origin.srcEid, toAddress, amountReceivedLD);
+        }
+    }
+
+    function _updateCirculatingSupply(address sender, uint256 amount) private {
```

Comment:
To confirm - this is only ever to update when burning, not to mint - is that correct?

Reply:
Yes, when burning @frontier159